### DISCUSSION TITLE
Serverside rendering

### DISCUSSION THREAD

**Discussion by akhilgangula** [vercel/next.js#21952] [Help] [answered]
I was trying to differentiate static Generation and server-side rendering techniques.
So I have created 2 pages for it, one with getServerSideProps for SSR and one without it, and placed console log in both.
```
const SSRTest=()=> {
 console.log('SSR: Reached here');  
...  
}  
export const getServerSideProps =() =>...   
export default SSRTest;  
`

`
const SGTest = () => {   
console.log('SG: Reached here');   
...   
}
export default SGTest
```
I was assuming that SSR should log in the terminal where the server is running and SG was supposed to log on the browser. Instead, I saw SSR log logged on the browser.

I am just wondering why the SSR log was logged on the browser. Any idea?

**Comment by michrome** [comment, marked as answer]
If your `console.log` method was in `getServerSideProps` for example then you’d see what you’re expecting. 

A server rendered page can still contain JavaScript that’s run by the browser. And don’t forget even statically generated pages are still reactive: Next.js will hydrate your application client-side to give it full interactivity.

**Reply by akhilgangula** [reply]
Thanks that made sense.
So in case of Server side rendering, next Js will not send response as HTML but UI will be rendering on client ? 
When I load page initially the response of the request should be HTML and when I switch the page via clicking a link, response is in json, is it next Js way to optimize performance ?

**Reply by michrome** [reply]
Server Side Rendering means the response sent by the server **_is_** a pre-rendered HTML page. The pre-rendered HTML page includes JavaScript code which runs and makes the page fully interactive (this process is called hydration.)
